src/framework/application.js

```diff
@@ -154,7 +159,15 @@ class Application extends AppBase {
         }
         options.graphicsDeviceOptions.alpha = options.graphicsDeviceOptions.alpha || false;
 
-        return new WebglGraphicsDevice(canvas, options.graphicsDeviceOptions);
+        const device = new WebglGraphicsDevice(canvas, options.graphicsDeviceOptions);
+
+        // register shader programs
+        device.programLib.register('basic', basic);
+        device.programLib.register('particle', particle);
+        device.programLib.register('skybox', skybox);
+        device.programLib.register('standard', standard);
```

Comment:
So this presumably means that anyone using AppBase now also needs to explicitly register shader programs?

Reply:
Correct. It also becomes tree-shakable.

Reply:
At the moment this PR would not allow BaseApp to be used instead of the Application (which it is currently) as those generators (basic, particle ..) are not exported and so cannot be from outside set on the device.

Reply:
what would you suggest we do instead?

Reply:
Maybe those lines should be moved to AppBase till we're ready to deal with it?

Reply:
Yep makes sense. Moved to AppBase.init. Thanks!